include/swift/AST/IRGenOptions.h

```diff
@@ -168,6 +173,7 @@ class IRGenOptions {
     Hash = (Hash << 1) | Optimize;
     Hash = (Hash << 1) | DisableLLVMOptzns;
     Hash = (Hash << 1) | DisableLLVMARCOpts;
+    Hash = (Hash << 2) | (unsigned)Sanitize;
```

Comment:
This computes the hash of everything *but* the IR. The optimization settings are things that happen to the IR after it's been emitted, but ASan's extra attributes should mean it's already factored in.